src/main/java/seedu/address/model/client/Address.java

```diff
@@ -34,14 +34,18 @@ public Address(String address) {
      * Returns true if a given string is a valid address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) | test == "";
```

Comment:
Conflicts with `MESSAGE_CONSTRAINT` that it can be blank. `isValidAddress` is somehow (when you instantiate an empty Email in line 44 of `AddClientCommandParser`) still called even though shouldn't be called. If no addresses were provided, then this check won't be needed.

This is tricky to solve without going in depth into model, storage etc. A suggestion for a simple fix is to change the definition of validAddress, Email etc, and reflect them in their `MESSAGE_CONSTRAINT`, and header comments if any defintions related to it occurs..

Reply:
Can you illustrate an example where it causes a bug? I can't seem to produce any bugs on my end. It would be helpful if you can show me the bug and your expected output so I can have a better idea on your suggestion

Reply:
This is actually not a software bug, but rather a code quality issue, bc in message constraints we say that address cannot be blank, but the `isValidAddress` explicitly says that it allows blank addresses. Furthermore the parser gets more coupled with isValidAddress. Can lead to confusion.